api/Controllers/AccountsController.cs

```diff
@@ -12,24 +14,29 @@ namespace iTechArt.Hotels.Api.Controllers
     public class AccountsController : Controller
     {
         private readonly HotelsDatabaseContext _hotelsDb;
+        private readonly IHashPasswords _hashPasswordsService;
 
-        public AccountsController(HotelsDatabaseContext hotelsDb)
+        public AccountsController(HotelsDatabaseContext hotelsDb, IHashPasswords hashPasswordsService)
         {
             _hotelsDb = hotelsDb;
+            _hashPasswordsService = hashPasswordsService;
         }
 
         [HttpPost]
-        [Authorize]
+        [Authorize(Roles = "admin")]
         public async Task<IActionResult> CreateAccount([FromBody] Login request)
         {
             if (!CheckIfEmailUnique(request.Email))
             {
                 return BadRequest("User is already registered with this email");
             }
+            byte[] salt = _hashPasswordsService.GenerateSalt();
             var user = new Account
             {
                 Email = request.Email,
-                Password = request.Password
+                Salt = Convert.ToBase64String(salt),
+                Password = _hashPasswordsService.HashPassword(request.Password, salt),
+                Role = "admin"
```

Comment:
Is there any way to avoid the use of inline role string?
You could make a typo, and it would be pretty hard to figure out why something is not working correctly :)